pkg/gather/clusterconfig/gather_logs.go

```diff
@@ -16,26 +16,50 @@ import (
 	"github.com/openshift/insights-operator/pkg/record"
 )
 
-// gatherLogsFromPodsInNamespace collects logs from the pods in provided namespace
-//   - messagesToSearch are the messages to filter the logs(case-insensitive)
-//   - sinceSeconds sets the moment to fetch logs from(current time - sinceSeconds)
-//   - limitBytes sets the maximum amount of logs that can be fetched
-//   - logFileName sets the name of the file to save logs to.
-//   - labelSelector allows you to filter pods by their labels
-//   - regexSearch makes messagesToSearch regex patterns, so you can accomplish more complicated search
+type logsContainersFilter struct {
+	namespace                string
+	labelSelector            string
+	containerNameRegexFilter string
+}
+
+type logMessagesFilter struct {
+	messagesToSearch []string
+	regexSearch      bool
+	sinceSeconds     int64
+	limitBytes       int64
+}
+
+// gatherLogsFromContainers collects logs from containers
+//   - containerFilter allows you to specify
+//     - namespace in which to search for pods
+//     - labelSelector to filter pods by their labels (keep empty to not filter)
+//     - containerNameRegexFilter to filter containers in the pod (keep empty to not filter)
+//   - logMessagesFilter allows you to specify
+//     - messagesToSearch to filter the logs by substrings (case-insensitive)
+//       or regex (add `(?i)` in the beginning to make search case-insensitive)
+//     - regexSearch which makes messagesToSearch regex patterns, so you can accomplish more complicated search
+//     - sinceSeconds which sets the moment to fetch the logs from (current time - sinceSeconds)
+//     - limitBytes which sets the maximum amount of logs that can be fetched
+//   - logFileName sets the name of the file to save the logs to.
 //
 // Location of the logs is `config/pod/{namespace}/logs/{podName}/{fileName}.log`
-func gatherLogsFromPodsInNamespace(
+func gatherLogsFromContainers(
 	ctx context.Context,
 	coreClient v1.CoreV1Interface,
-	namespace string,
-	messagesToSearch []string,
-	sinceSeconds int64,
-	limitBytes int64,
+	containersFilter logsContainersFilter,
+	logMessagesFilter logMessagesFilter,
 	logFileName string,
-	labelSelector string,
-	regexSearch bool,
 ) ([]record.Record, error) {
+	var (
+		namespace                = containersFilter.namespace
+		labelSelector            = containersFilter.labelSelector
+		containerNameRegexFilter = containersFilter.containerNameRegexFilter
+		messagesToSearch         = logMessagesFilter.messagesToSearch
+		regexSearch              = logMessagesFilter.regexSearch
+		sinceSeconds             = logMessagesFilter.sinceSeconds
+		limitBytes               = logMessagesFilter.limitBytes
```

Comment:
Is this blocked really needed here? Maybe I would use some shorter name for the local variables and then use them directly.

Reply:
fixed